specification.md

```diff
@@ -3657,30 +3624,46 @@ decide to strictly enforce it.
 
 ### Workflow Error Handling
 
-Any state can encounter runtime errors. Errors can arise from state failures such as exceptions thrown during function
-execution, network errors, or errors in the workflow definition (e.g., incorrect paths).
-If a runtime error is not explicitly handled within the state definition, the default course of action should be to
-halt workflow execution.
+Serverless Workflow language allows you to define `explicit` error handling, meaning you can define what should happen
+in case of errors inside your workflow model rather than some generic error handling entity.
+This allows error handling to become part of your orchestration activities and as such part of your business problem
+solutions.
 
-In the case of runtime errors, implementations should expose these errors to the workflow state via the "error" JSON object.
-This object should include properties "name", "message", and "trace".
-Here is an example of such error object:
+Each workflow state can define error handling, which is related only to errors that may arise during its 
+execution. Error handling defined in one state cannot be used to handle errors that happened during execution of another state
+during workflow execution.
 
-```json
-{
-  "error": {
-    "name": "FunctionExecutionError",
-     "message": "Write permission denied on file: '/xxx/yyy'",
-     "trace": "Exception in thread 'main' java.lang.RuntimeException:\nat com.MyFunction.methodXYZ(MyFunction.java:13)"
-  }
-}
-```
+Errors that may arise during workflow execution that are not explicitly handled within the workflow definition
+should be reported by runtime implementations and halt workflow execution, 
+
+Within workflow definitions, errors defined are `domain specific`, meaning they are defined within 
+the actual business domain, rather than their technical (programming-language-specific) description.
 
-Each state can explicitly catch runtime errors via its `onError` property. This property includes one or more
-definitions matching the error object properties and defining a transition to a workflow state representing the
-workflow execution flow in case of that particular error.
+For example, we can define errors such as "Order not found", or "Item not in inventory", rather than having to
+use terms such as "java.lang.IllegalAccessError", or "response.status == 404", which 
+might make little to no sense to our specific problem domain, as well as may not be portable across various runtime implementations.
+
+In addition to the domain specific error name, users have the option to also an optional error code
+to help runtime implementations with mapping defined errors to concrete underlying technical ones.
```

Comment:
I don't think it makes sense to allow specifying `code` if the name is `*`.  Should we call that out here that if name==`*`, then `code` must be left undefined?

Reply:
agreed. i will add this a little bit lower in the "Defining Errors" section and also will try to update the schema to catch this case during validation